In [1]:
import json
import networkx as nx
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import os

## Analysis of the Environmental Factors that influence our model

#### First we will read the excel file containing the environmental datasheet.

#### taking only the reversed variables (factors) in the dataframe, they are needed for the analysis

In [2]:
env=pd.read_csv('./data/environment.csv', sep=';', header=0)

In [3]:
env = env[['Child_Bosse','School', 'Class','Wave','GEN_FAS_computer_R','GEN_FAS_car_R',
           'GEN_FAS_vacation_R','GEN_FAS_ownroom_R']]

In [5]:
env.head()

Child_Bosse  School  Class  Wave  GEN_FAS_computer_R  GEN_FAS_car_R  \
0          643      22   52.0     2                 NaN            NaN   
1          643      22   52.0     4                 NaN            NaN   
2          643      22   52.0     3                 NaN            NaN   
3          643      22   52.0     1                 NaN            NaN   
4          645      22   52.0     2                 NaN            NaN   

   GEN_FAS_vacation_R  GEN_FAS_ownroom_R  
0                 NaN                NaN  
1                 NaN                NaN  
2                 NaN                NaN  
3                 NaN                NaN  
4                 NaN                NaN

In [6]:
classes=[67, 71, 72, 74, 77, 78, 79, 81, 83, 86, 100, 101, 103, 121, 122, 125, 126, 127, 129, 130, 131, 133, 135, 136, 138, 139]

#### take only the rows with the particular classes

In [7]:
env = env[env['Class'].isin(classes)]

In [8]:
env.head()

Child_Bosse  School  Class  Wave  GEN_FAS_computer_R  GEN_FAS_car_R  \
420          963      25   67.0     1                 NaN            NaN   
421          963      25   67.0     3                 NaN            NaN   
422          963      25   67.0     4                 NaN            NaN   
423          963      25   67.0     2                 NaN            NaN   
424          965      25   67.0     1                 0.0            1.0   

     GEN_FAS_vacation_R  GEN_FAS_ownroom_R  
420                 NaN                NaN  
421                 NaN                NaN  
422                 NaN                NaN  
423                 NaN                NaN  
424                 3.0                0.0

### looking at the missing values

In [9]:
env.isnull().sum()

Child_Bosse             0
School                  0
Class                   0
Wave                    0
GEN_FAS_computer_R    400
GEN_FAS_car_R         400
GEN_FAS_vacation_R    400
GEN_FAS_ownroom_R     400
dtype: int64

In [10]:
env.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1840 entries, 420 to 3707
Data columns (total 8 columns):
Child_Bosse           1840 non-null int64
School                1840 non-null int64
Class                 1840 non-null float64
Wave                  1840 non-null int64
GEN_FAS_computer_R    1440 non-null float64
GEN_FAS_car_R         1440 non-null float64
GEN_FAS_vacation_R    1440 non-null float64
GEN_FAS_ownroom_R     1440 non-null float64
dtypes: float64(5), int64(3)
memory usage: 129.4 KB


### missing values in percentage

In [11]:
(1 - (len(env)-env.isnull().sum()) / len(env))*100

Child_Bosse            0.00000
School                 0.00000
Class                  0.00000
Wave                   0.00000
GEN_FAS_computer_R    21.73913
GEN_FAS_car_R         21.73913
GEN_FAS_vacation_R    21.73913
GEN_FAS_ownroom_R     21.73913
dtype: float64

### number of kids that have no data input, at all.

In [12]:
all_missing_e = env

In [13]:
all_missing_e = all_missing_e.groupby('Child_Bosse')

In [14]:
df_help_missing_e = all_missing_e.sum()

In [15]:
df_help_missing_e

School  Class  Wave  GEN_FAS_computer_R  GEN_FAS_car_R  \
Child_Bosse                                                           
963             100  268.0    10                 0.0            0.0   
965             100  268.0    10                 0.0            4.0   
966             100  268.0    10                 0.0            0.0   
967             100  268.0    10                 0.0            4.0   
968             100  268.0    10                 0.0            4.0   
971             100  268.0    10                 0.0            4.0   
972             100  268.0    10                 4.0            0.0   
973             100  268.0    10                 0.0            0.0   
974             100  268.0    10                 0.0            0.0   
975             100  268.0    10                 0.0            8.0   
977             100  268.0    10                 4.0            4.0   
978             100  268.0    10                 0.0            4.0   
979             100  268.0    10                 0.0            4.0   
980             100  268.0    10                 0.0            4.0   
982             100  268.0    10                 0.0            0.0   
983             100  268.0    10                 0.0            0.0   
985             100  268.0    10                 0.0            0.0   
987             100  268.0    10                 0.0            0.0   
1083            100  284.0    10                 0.0            0.0   
1084            100  284.0    10                 0.0            0.0   
1085            100  284.0    10                 0.0            0.0   
1086            100  284.0    10                 0.0            4.0   
1087            100  284.0    10                 0.0            4.0   
1090            100  284.0    10                 4.0            4.0   
1093            100  284.0    10                 0.0            4.0   
1094            100  284.0    10                 0.0            0.0   
1095            100  284.0    10                 0.0            0.0   
1097            100  284.0    10                 0.0            8.0   
1098            100  284.0    10                 0.0            0.0   
1099            100  284.0    10                 0.0            4.0   
...             ...    ...   ...                 ...            ...   
2921            160  540.0    10                 0.0            0.0   
2922            160  540.0    10                 0.0            0.0   
2923            160  540.0    10                 8.0            8.0   
2924            160  540.0    10                 0.0            0.0   
2925            160  540.0    10                 8.0            0.0   
2926            160  540.0    10                 0.0            0.0   
2927            160  540.0    10                 0.0            4.0   
2928            160  540.0    10                 0.0            0.0   
2929            160  540.0    10                 0.0            0.0   
2930            160  540.0    10                 0.0            0.0   
2931            160  540.0    10                 0.0            4.0   
2932            160  540.0    10                 0.0            0.0   
2933            160  540.0    10                 4.0            4.0   
2999            160  524.0    10                 0.0            0.0   
3000            160  540.0    10                 0.0            0.0   
3001            160  532.0    10                 0.0            0.0   
3002            160  532.0    10                 0.0            4.0   
3003            160  532.0    10                 4.0            0.0   
3004            160  532.0    10                 0.0            4.0   
3005            160  532.0    10                 0.0            0.0   
3058            132  400.0    10                 0.0            0.0   
4621            160  540.0    10                 0.0            0.0   
4622            160  540.0    10                 0.0            0.0   
4625            160  524.0    10     

In [15]:
df_help_missing_e=df_help_missing_e.drop(['School', 'Class','Wave'], axis=1)

In [16]:
df_help_missing_e

GEN_FAS_computer_R  GEN_FAS_car_R  GEN_FAS_vacation_R  \
Child_Bosse                                                          
963                         NaN            NaN                 NaN   
965                         0.0            4.0                12.0   
966                         0.0            0.0                 0.0   
967                         0.0            4.0                 4.0   
968                         0.0            4.0                12.0   
971                         0.0            4.0                 4.0   
972                         4.0            0.0                 4.0   
973                         NaN            NaN                 NaN   
974                         0.0            0.0                 0.0   
975                         0.0            8.0                12.0   
977                         4.0            4.0                 4.0   
978                         0.0            4.0                12.0   
979                         0.0            4.0                 8.0   
980                         0.0            4.0                12.0   
982                         NaN            NaN                 NaN   
983                         0.0            0.0                 0.0   
985                         NaN            NaN                 NaN   
987                         0.0            0.0                 0.0   
1083                        NaN            NaN                 NaN   
1084                        0.0            0.0                12.0   
1085                        0.0            0.0                 4.0   
1086                        0.0            4.0                12.0   
1087                        0.0            4.0                 4.0   
1090                        4.0            4.0                 8.0   
1093                        0.0            4.0                 0.0   
1094                        0.0            0.0                 8.0   
1095                        0.0            0.0                 8.0   
1097                        0.0            8.0                 0.0   
1098                        NaN            NaN                 NaN   
1099                        0.0            4.0                 8.0   
...                         ...            ...                 ...   
2921                        0.0            0.0                 0.0   
2922                        0.0            0.0                 4.0   
2923                        8.0            8.0                 4.0   
2924                        0.0            0.0                 8.0   
2925                        8.0            0.0                 8.0   
2926                        0.0            0.0                 4.0   
2927                        0.0            4.0                 4.0   
2928                        0.0            0.0                12.0   
2929                        0.0            0.0                 8.0   
2930                        0.0            0.0                 4.0   
2931                        0.0            4.0                 8.0   
2932                        0.0            0.0                 0.0   
2933                        4.0            4.0                 8.0   
2999                        0.0            0.0                 0.0   
3000                        0.0            0.0                 0.0   
3001                        0.0            0.0                 4.0   
3002                        0.0            4.0                 4.0   
3003                        4.0            0.0                 0.0   
3004                        0.0            4.0                 4.0   
3005                        0.0            0.0                 4.0   
3058                        NaN            NaN                 NaN   
4621                        NaN            NaN                 NaN   
4622                        NaN            NaN                 NaN   
4625                        NaN            NaN                 NaN   
4626                   

In [17]:
df_only_missing=df_help_missing_e[df_help_missing_e.isnull().sum(1)==4]

In [18]:
df_only_missing

GEN_FAS_computer_R  GEN_FAS_car_R  GEN_FAS_vacation_R  \
Child_Bosse                                                          
963                         NaN            NaN                 NaN   
973                         NaN            NaN                 NaN   
982                         NaN            NaN                 NaN   
985                         NaN            NaN                 NaN   
1083                        NaN            NaN                 NaN   
1098                        NaN            NaN                 NaN   
1101                        NaN            NaN                 NaN   
1105                        NaN            NaN                 NaN   
1107                        NaN            NaN                 NaN   
1119                        NaN            NaN                 NaN   
1120                        NaN            NaN                 NaN   
1123                        NaN            NaN                 NaN   
1126                        NaN            NaN                 NaN   
1134                        NaN            NaN                 NaN   
1137                        NaN            NaN                 NaN   
1155                        NaN            NaN                 NaN   
1157                        NaN            NaN                 NaN   
1158                        NaN            NaN                 NaN   
1160                        NaN            NaN                 NaN   
1168                        NaN            NaN                 NaN   
1278                        NaN            NaN                 NaN   
1292                        NaN            NaN                 NaN   
1348                        NaN            NaN                 NaN   
1355                        NaN            NaN                 NaN   
1468                        NaN            NaN                 NaN   
1469                        NaN            NaN                 NaN   
1554                        NaN            NaN                 NaN   
1564                        NaN            NaN                 NaN   
2007                        NaN            NaN                 NaN   
2010                        NaN            NaN                 NaN   
...                         ...            ...                 ...   
2776                        NaN            NaN                 NaN   
2777                        NaN            NaN                 NaN   
2778                        NaN            NaN                 NaN   
2779                        NaN            NaN                 NaN   
2780                        NaN            NaN                 NaN   
2782                        NaN            NaN                 NaN   
2783                        NaN            NaN                 NaN   
2834                        NaN            NaN                 NaN   
2835                        NaN            NaN                 NaN   
2840                        NaN            NaN                 NaN   
2844                        NaN            NaN                 NaN   
2845                        NaN            NaN                 NaN   
2848                        NaN            NaN                 NaN   
2850                        NaN            NaN                 NaN   
2852                        NaN            NaN                 NaN   
2859                        NaN            NaN                 NaN   
2864                        NaN            NaN                 NaN   
2903                        NaN            NaN                 NaN   
2906                        NaN            NaN                 NaN   
2914                        NaN            NaN                 NaN   
3058                        NaN            NaN                 NaN   
4621                        NaN            NaN                 NaN   
4622                        NaN            NaN                 NaN   
4625                        NaN            NaN                 NaN   
4626                   

In [19]:
(1-(df_help_missing_e.shape[0]-df_only_missing.shape[0]) / df_help_missing_e.shape[0]) * 100

21.739130434782606

In [20]:
df_help_missing_e.shape[0]

460

In [21]:
df_only_missing.shape[0]

100

### conclusion: we have no data whatsoever for 100 out of 460 kids. 21.73% of completely missing data

In [22]:
missing_users_index=df_only_missing.index.values

In [23]:
missing_users_index

array([ 963,  973,  982,  985, 1083, 1098, 1101, 1105, 1107, 1119, 1120,
       1123, 1126, 1134, 1137, 1155, 1157, 1158, 1160, 1168, 1278, 1292,
       1348, 1355, 1468, 1469, 1554, 1564, 2007, 2010, 2019, 2020, 2021,
       2051, 2568, 2583, 2588, 2590, 2592, 2627, 2628, 2631, 2652, 2658,
       2662, 2664, 2669, 2670, 2671, 2672, 2675, 2678, 2680, 2698, 2701,
       2710, 2711, 2720, 2758, 2759, 2760, 2761, 2762, 2764, 2765, 2767,
       2768, 2771, 2773, 2774, 2776, 2777, 2778, 2779, 2780, 2782, 2783,
       2834, 2835, 2840, 2844, 2845, 2848, 2850, 2852, 2859, 2864, 2903,
       2906, 2914, 3058, 4621, 4622, 4625, 4626, 4627, 4628, 4629, 4630,
       4631])

In [24]:
users_with_data_e = e[~e['Child_Bosse'].isin(missing_users_index.tolist())]

In [25]:
users_with_data_e

Child_Bosse  School  Class  Wave  GEN_FAS_computer_R  GEN_FAS_car_R  \
424           965      25   67.0     1                 0.0            1.0   
425           965      25   67.0     4                 0.0            1.0   
426           965      25   67.0     2                 0.0            1.0   
427           965      25   67.0     3                 0.0            1.0   
428           966      25   67.0     3                 0.0            0.0   
429           966      25   67.0     1                 0.0            0.0   
430           966      25   67.0     4                 0.0            0.0   
431           966      25   67.0     2                 0.0            0.0   
432           967      25   67.0     2                 0.0            1.0   
433           967      25   67.0     1                 0.0            1.0   
434           967      25   67.0     4                 0.0            1.0   
435           967      25   67.0     3                 0.0            1.0   
436           968      25   67.0     4                 0.0            1.0   
437           968      25   67.0     3                 0.0            1.0   
438           968      25   67.0     1                 0.0            1.0   
439           968      25   67.0     2                 0.0            1.0   
440           971      25   67.0     3                 0.0            1.0   
441           971      25   67.0     4                 0.0            1.0   
442           971      25   67.0     2                 0.0            1.0   
443           971      25   67.0     1                 0.0            1.0   
444           972      25   67.0     2                 1.0            0.0   
445           972      25   67.0     3                 1.0            0.0   
446           972      25   67.0     4                 1.0            0.0   
447           972      25   67.0     1                 1.0            0.0   
452           974      25   67.0     3                 0.0            0.0   
453           974      25   67.0     2                 0.0            0.0   
454           974      25   67.0     4                 0.0            0.0   
455           974      25   67.0     1                 0.0            0.0   
456           975      25   67.0     1                 0.0            2.0   
457           975      25   67.0     2                 0.0            2.0   
...           ...     ...    ...   ...                 ...            ...   
3650         2841      42  139.0     2                 0.0            0.0   
3651         2841      42  139.0     4                 0.0            0.0   
3652         2843      42  139.0     1                 0.0            0.0   
3653         2843      42  139.0     2                 0.0            0.0   
3654         2843      42  139.0     4                 0.0            0.0   
3655         2843      42  139.0     3                 0.0            0.0   
3668         2849      42  139.0     1                 0.0            0.0   
3669         2849      42  139.0     2                 0.0            0.0   
3670         2849      42  139.0     3                 0.0            0.0   
3671         2849      42  139.0     4                 0.0            0.0   
3680         2853      42  139.0     2                 0.0            1.0   
3681         2853      42  139.0     4                 0.0            1.0   
3682         2853      42  139.0     3                 0.0            1.0   
3683         2853      42  139.0     1                 0.0            1.0   
3684         2857      42  139.0     3                 0.0            0.0   
3685         2857      42  139.0     1                 0.0            0.0   
3686         2857      42  139.0     2                 0.0            0.0   
3687         2857      42  139.0     4                 0.0            0.0   
3688         2858      42  139.0     2                 0.0            0.0   
3689         2858      42  139.0     1                 0.0            0.0   
36

### we got the kids that have some data in the 'users_with_data_e' dataframe!

### Step1 let's check if there is difference in answers per kid, comparing the 4 waves

In [26]:
index_kids=users_with_data_e.Child_Bosse.unique().tolist()
len(index_kids)

360

we are working with 360 kids total

This 'for' cycle checks if there is a difference at waves, for each user. We want to check if the column values for the environmental factors changed or not.

In [27]:
user_list_same_waves_val = [] 
user_list_dif_waves_val = [] 
for user in index_kids:
    user_df=users_with_data_e[users_with_data_e.Child_Bosse == user]
    if(user_df.GEN_FAS_computer_R.nunique()==1 and user_df.GEN_FAS_car_R.nunique()==1 and user_df.GEN_FAS_vacation_R.nunique()==1 and user_df.GEN_FAS_ownroom_R.nunique()==1):
        user_list_same_waves_val.append(user)
    else:
        user_list_diff_waves_val.append(user)

In [28]:
len(user_list_same_waves_val)

360

In [29]:
len(user_list_dif_waves_val)

0

### all our users have same inputs in every wave, we can simply take wave one for all of them for the analysis!

In [30]:
final_e=users_with_data_e[users_with_data_e.Wave==1]

In [31]:
final_e.shape[0]

360

In [32]:
final_e

Child_Bosse  School  Class  Wave  GEN_FAS_computer_R  GEN_FAS_car_R  \
424           965      25   67.0     1                 0.0            1.0   
429           966      25   67.0     1                 0.0            0.0   
433           967      25   67.0     1                 0.0            1.0   
438           968      25   67.0     1                 0.0            1.0   
443           971      25   67.0     1                 0.0            1.0   
447           972      25   67.0     1                 1.0            0.0   
455           974      25   67.0     1                 0.0            0.0   
456           975      25   67.0     1                 0.0            2.0   
460           977      25   67.0     1                 1.0            1.0   
465           978      25   67.0     1                 0.0            1.0   
469           979      25   67.0     1                 0.0            1.0   
472           980      25   67.0     1                 0.0            1.0   
482           983      25   67.0     1                 0.0            0.0   
489           987      25   67.0     1                 0.0            0.0   
662          1084      25   71.0     1                 0.0            0.0   
665          1085      25   71.0     1                 0.0            0.0   
669          1086      25   71.0     1                 0.0            1.0   
673          1087      25   71.0     1                 0.0            1.0   
679          1090      25   71.0     1                 1.0            1.0   
680          1093      25   71.0     1                 0.0            1.0   
685          1094      25   71.0     1                 0.0            0.0   
689          1095      25   71.0     1                 0.0            0.0   
694          1097      25   71.0     1                 0.0            2.0   
702          1099      25   71.0     1                 0.0            1.0   
705          1100      25   71.0     1                 0.0            0.0   
713          1102      25   71.0     1                 0.0            0.0   
717          1104      25   71.0     1                 0.0            0.0   
730          1108      25   71.0     1                 0.0            0.0   
733          1109      25   71.0     1                 0.0            0.0   
739          1111      25   72.0     1                 1.0            1.0   
...           ...     ...    ...   ...                 ...            ...   
3366         2735      42  130.0     1                 0.0            0.0   
3555         2811      42  138.0     1                 0.0            0.0   
3557         2812      42  138.0     1                 0.0            1.0   
3561         2813      42  138.0     1                 0.0            0.0   
3567         2814      42  138.0     1                 0.0            0.0   
3571         2816      42  138.0     1                 0.0            0.0   
3573         2818      42  138.0     1                 0.0            0.0   
3576         2819      42  138.0     1                 0.0            2.0   
3582         2820      42  138.0     1                 0.0            0.0   
3585         2822      42  138.0     1                 1.0            1.0   
3588         2823      42  138.0     1                 0.0            0.0   
3594         2824      42  138.0     1                 0.0            1.0   
3596         2825      42  138.0     1                 0.0            1.0   
3603         2827      42  138.0     1                 0.0            0.0   
3606         2828      42  138.0     1                 0.0            0.0   
3608         2829      42  138.0     1                 0.0            0.0   
3613         2830      42  138.0     1                 0.0            0.0   
3617         2831      42  138.0     1                 0.0            0.0   
3621         2832      42  138.0     1                 0.0            0.0   
3626         2833      42  138.0     1                 0.0            1.0   
36

In [33]:
final_e.min(axis=0)

Child_Bosse           965.0
School                 25.0
Class                  67.0
Wave                    1.0
GEN_FAS_computer_R      0.0
GEN_FAS_car_R           0.0
GEN_FAS_vacation_R      0.0
GEN_FAS_ownroom_R       0.0
dtype: float64

In [34]:
final_e.max(axis=0)

Child_Bosse           3005.0
School                  42.0
Class                  139.0
Wave                     1.0
GEN_FAS_computer_R       3.0
GEN_FAS_car_R            2.0
GEN_FAS_vacation_R       3.0
GEN_FAS_ownroom_R        1.0
dtype: float64

**min max ranges for the environmental factors of interest**

**GEN_FAS_computer_R **      0,3

**GEN_FAS_car_R **           0,2

**GEN_FAS_vacation_R**       0,3

**GEN_FAS_ownroom_R**        0,1


In [35]:
col_list= list(['GEN_FAS_computer_R',
       'GEN_FAS_car_R', 'GEN_FAS_vacation_R',
       'GEN_FAS_ownroom_R'])

Append new FAS_Score column to the final_e dataframe, summing up the selected columns per row.

In [36]:
#final_e['FAS_Score']=final_e[col_list].sum(axis=1)

In [37]:
#final_e

In [38]:
#final_e=final_e[['Child_Bosse','FAS_Score']]

In [39]:
#final_e.set_index('Child_Bosse')

In [40]:
#final_e.FAS_Score.astype('int')

In [41]:
#final_e.FAS_Score.hist(bins=12,figsize=((16,8)))

In [43]:
#final_e.FAS_Score.describe()

In [54]:
ncol=final_e[col_list].sum(axis=1).to_frame()

In [59]:
ncol.columns=['FAS_Score_R']

In [71]:
ncol

FAS_Score_R
424           4.0
429           0.0
433           2.0
438           4.0
443           2.0
447           2.0
455           0.0
456           5.0
460           3.0
465           4.0
469           3.0
472           4.0
482           0.0
489           0.0
662           3.0
665           1.0
669           4.0
673           2.0
679           4.0
680           1.0
685           2.0
689           2.0
694           2.0
702           3.0
705           2.0
713           0.0
717           3.0
730           2.0
733           1.0
739           4.0
...           ...
3366          2.0
3555          2.0
3557          2.0
3561          2.0
3567          2.0
3571          2.0
3573          1.0
3576          4.0
3582          0.0
3585          3.0
3588          0.0
3594          2.0
3596          2.0
3603          0.0
3606          0.0
3608          1.0
3613          3.0
3617          0.0
3621          1.0
3626          4.0
3639          3.0
3640          2.0
3648          1.0
3652          2.0
3668          0.0
3683          2.0
3685          1.0
3689          0.0
3699          3.0
3702          6.0

[360 rows x 1 columns]

In [72]:
final_e = pd.concat([final_e, ncol], axis=1)

In [74]:
 final_e=final_e[['Child_Bosse','FAS_Score_R']]

In [75]:
final_e

Child_Bosse  FAS_Score_R
424           965          4.0
429           966          0.0
433           967          2.0
438           968          4.0
443           971          2.0
447           972          2.0
455           974          0.0
456           975          5.0
460           977          3.0
465           978          4.0
469           979          3.0
472           980          4.0
482           983          0.0
489           987          0.0
662          1084          3.0
665          1085          1.0
669          1086          4.0
673          1087          2.0
679          1090          4.0
680          1093          1.0
685          1094          2.0
689          1095          2.0
694          1097          2.0
702          1099          3.0
705          1100          2.0
713          1102          0.0
717          1104          3.0
730          1108          2.0
733          1109          1.0
739          1111          4.0
...           ...          ...
3366         2735          2.0
3555         2811          2.0
3557         2812          2.0
3561         2813          2.0
3567         2814          2.0
3571         2816          2.0
3573         2818          1.0
3576         2819          4.0
3582         2820          0.0
3585         2822          3.0
3588         2823          0.0
3594         2824          2.0
3596         2825          2.0
3603         2827          0.0
3606         2828          0.0
3608         2829          1.0
3613         2830          3.0
3617         2831          0.0
3621         2832          1.0
3626         2833          4.0
3639         2838          3.0
3640         2839          2.0
3648         2841          1.0
3652         2843          2.0
3668         2849          0.0
3683         2853          2.0
3685         2857          1.0
3689         2858          0.0
3699         2862          3.0
3702         2863          6.0

[360 rows x 2 columns]